test/test_jit.py

```diff
@@ -10481,6 +10481,46 @@ def t(x):
 
         self.checkScript(t, (torch.zeros(3, 2, 3),))
 
+    def test_torch_ignore_conversion_to_none(self):
+        class A(torch.nn.Module):
+            def __init__(self):
+                super(A, self).__init__()
+
+            @torch.jit.ignore
+            def ignored(self, a: int) -> None:
+                l: int = len([2 for i in range(a) if i > 2])
+                return
+
+            def forward(self) -> int:
+                a: int = 4
+                b: int = 5
+                self.ignored(a)
+                return a + b
+
+        class B(torch.nn.Module):
+            def __init__(self):
+                super(B, self).__init__()
+
+            @torch.jit.ignore
+            def ignored(self, a: int):
```

Comment:
How does TS grab the return type for a script-able function these days? Does it assume it's a `Tensor` or does it try to pull out the return type from the statements in the body? This should follow the same rules as that (i.e. if no return is assumed to be `Tensor`, adding a special rule for `torch.jit.ignore` seems weird)

Reply:
I think whenever it sees the annotation is NoneType, it calls TensorType.getInferred() to annotate it as Tensor. (https://github.com/pytorch/pytorch/blob/10abbb812a47bfabba4905b50dde02a511191522/torch/jit/annotations.py#L297)